### DISCUSSION TITLE
How to get SOLVERTIME from the OPTIMIZER?

### DISCUSSION THREAD

**Discussion by oravec-juraj** [yalmip/YALMIP#1398] [General discussion] [answered]
Dear Johan,

   please, is there an effective way to extract the pure value of the SOLVERTIME from the YALMIP object OPTIMIZER? 
We found a line of code in the "OPTIMIZER.M" function that seems to manage to disable its evaluation:

    model.getsolvertime = 0;

but we would appreciate accessing that value.
Thank you very much in advance. 
Best regards,
Juraj

**Comment by johanlofberg** [comment]
Not possible without hacking, turned off to eliminate overhead of timing call

**Reply by oravec-juraj** [reply]
Dear Johan,
thank you very much for your fast response. Of course, it makes sense not to overload the evaluation of OPTIMIZER as it serves to minimize the runtimes. However, we do a specific tuning, where we would benefit from accessing the solvertime. Please, could you give us any hint---roughly navigate us where to start "hacking" the OPTIMIZER function to catch the SOLVERTIME for QP? 
Best regards,
Juraj

**Reply by johanlofberg** [reply]
Just uncommenting the line you found should be sufficient I think.

I will think of a way to control it by options

**Reply by oravec-juraj** [reply]
Dear Johan,
thank you, it would be great to have such an option in some future release of YALMIP!
Meanwhile, it was not sufficient for us just to comment/disable the setting:

    model.getsolvertime = 0;

as the OPTIMIZER function itself did not evaluate the SOLVERTIME. We would like to catch ("measure") its value, but we failed to find the exact piece of code executing/calling the solver. It would be helpful if you could navigate us, to where the QP solver is called, but, of course, we understand, if it is not that trivial to answer without deep diving into the code. 
Thank you very much in advance. 
Best regards,
Juraj

**Reply by johanlofberg** [reply, marked as answer]
Works here so not sure why it wouldn't on your machine. What solver?

```
>> P = optimizer(x<=y, -x,[],y,x);
>> [val,~,~,~,~,diagnostics] = P(4);
>> diagnostics

diagnostics = 

  struct with fields:

          Primal: [3×1 double]
            Dual: 1
           Slack: []
         problem: 0
         infostr: 'Successfully solved (LINPROG-2019.1)'
     solverinput: []
    solveroutput: []
      solvertime: 0.1366
```

**Reply by oravec-juraj** [reply]
Dear Johan,
thank you very much for fixing our problem! 
Previously, we did not access the SOLVERTIME, as we were not aware of returning DIAGNOSTICS as the 5th output of the OPTIMIZER (we expected just OPTIMAL_VALUE and ERROR_CODE on the output from the OPTIMIZER), therefore we tried to "manually" evaluate the corresponding SOLVERTIME. 
Thank you once again!
Best regards,
Juraj
